File: src/Application Specific/Forms And Controls/PartsToBeCredited/FRMAddPartToBeCredited.vb
Public Class FRMAddPartToBeCredited : Inherits FSM.FRMBaseForm : Implements IForm

    Public Sub New(ByVal qty As Integer)
        MyBase.New()

        InitializeComponent()
        Me.txtQtyAvailable.Text = qty
    End Sub

#Region "Interface Methods"

    Private Sub LoadMe(ByVal sender As Object, ByVal e As System.EventArgs) Implements IForm.LoadMe
        LoadForm(sender, e, Me)
    End Sub

    Public ReadOnly Property LoadedControl() As IUserControl Implements IForm.LoadedControl
        Get
            Return Nothing
        End Get
    End Property

    Private Sub ResetMe(ByVal newID As Integer) Implements IForm.ResetMe
    End Sub

#End Region
 
    Private Sub btnOK_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnOK.Click
        If Entity.Sys.Helper.MakeIntegerValid(txtQtyAvailable.Text) < Entity.Sys.Helper.MakeIntegerValid(txtQtyToReturn.Text) Then
            ShowMessage("Cannot return more than was allocated", MessageBoxButtons.OK, MessageBoxIcon.Information)
            Exit Sub
        End If
        Me.DialogResult =DialogResult.OK
    End Sub

End Class

File: src/Application Specific/Forms And Controls/PartsToBeCredited/FRMAddPartToBeCredited.cs
using System;
using System.Windows.Forms;

namespace FSM
{
    public partial class FRMAddPartToBeCredited : FRMBaseForm, IForm
    {
        public FRMAddPartToBeCredited()
        {
            InitializeComponent();
        }

        public FRMAddPartToBeCredited(int qty) : base()
        {
            InitializeComponent();
            txtQtyAvailable.Text = qty.ToString();
        }

        /* TODO ERROR: Skipped RegionDirectiveTrivia */

        public void LoadMe(object sender, EventArgs e)
        {
            LoadForm(sender, e, this);
        }

        public IUserControl LoadedControl
        {
            get
            {
                return null;
            }
        }

        public void ResetMe(int newID)
        {
        }

        /* TODO ERROR: Skipped EndRegionDirectiveTrivia */

        private void btnOK_Click(object sender, EventArgs e)
        {
            if (Entity.Sys.Helper.MakeIntegerValid(txtQtyAvailable.Text) < Entity.Sys.Helper.MakeIntegerValid(txtQtyToReturn.Text))
            {
                App.ShowMessage("Cannot return more than was allocated", MessageBoxButtons.OK, MessageBoxIcon.Information);
                return;
            }

            DialogResult = DialogResult.OK;
        }
    }
}